### DISCUSSION TITLE
Sponsored Brands - Attributed Purchases

### DISCUSSION THREAD

**Discussion by rodneyo** [amzn/ads-advanced-tools-docs#238] [Q&A] [answered]
Hi,
I need to create the report shown in the screen shot, created using the Amazon Ads Reporing console, using the advertising api.  However I can not find any mention or configuration for report_type_id of sbAttributedPurchases.  The report generated from the reporting console contains the promotedSku and promotedAsin.  I am using v3 endpoint for reporting.   Report configuration: 

```php
 'sponsored_brands-attributed-purchases' => [
                                'lookback' => '-14 day',
                                'report_name' => 'Weekly Data Pull - Amazon Product ASIN SB.csv',
                                'report_type_id' => 'sbAttributedPurchase',  ----> causes a 400 Bad Request 
                                'time_unit' => 'DAILY',
                                'format' => 'GZIP_JSON',
                                'ad_product' => 'SPONSORED_BRANDS',
                                'group_by' => ['purchasedAsin'],
                                'columns' => [
                                    "date" => "date",
                                    "campaignBudgetCurrencyCode" => "campaignBudgetCurrencyCode",
                                    "campaignId" => "campaignId",
                                    "campaignName" => "campaignName",
                                    "adGroupId" => "adGroupId",
                                    "adGroupName" => "adGroupName",
                                    "productName" => "productName",
                                    "attributionType" => "attributionType",
                                    //'promotedSku' => 'promotedSku',
                                    //'promotedAsin' => 'promotedAsin',
                                    "purchasedAsin" => "purchasedAsin",
                                    "sales14d" => "sales14d",
                                    "orders14d" => "orders14d",
                                    "unitsSold14d" => "unitsSold14d"
                                ]
                            ]
```

Report screen shot from reporting console.

![SponsoredBrands-AttributedPurchases](https://github.com/amzn/ads-advanced-tools-docs/assets/1626333/71bcc7d8-b4fc-4bb6-b2ce-c6606299c45d)

any help would be greatly     ##@@appreciated

**Comment by elizafarley** [comment]
Hi @rodneyo,

In the API, there is not a single report that can pull purchase data attributed to a single ASIN for Sponsored Brands. 

Instead you have two options:

1) [SB ads report](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v3/report-types/ad) - returns clicks/purchases/views at the adId level.
2) [SB purchased product report](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v3/report-types/purchased-product) - returns both promoted and brand halo ASINs that were purchased after an ad click or view.

Because there can be zero to many ASINs featured in an SB ad, it's difficult to attribute a purchase back to a single ASIN.

**Comment by rodneyo** [comment]
Hi eliza,
thanks for you help with this.  If I am understanding correctly.  There is no way to produce the type of report attached from the API?
['Weekly Data Pull - Amazon Product ASIN SB-Cosole.csv](https://github.com/amzn/ads-advanced-tools-docs/files/14845265/Weekly.Data.Pull.-.Amazon.Product.ASIN.SB-Cosole.csv)
This report was created from reporting console with the selection criteria from the screen shot I posted.    If open this report you'll notice the purchased ASIN.  Since you mentioned there is no report that can pull purchase data attributed to a single ASIN for SB.  How is the purchased ASIN being populated in this report?  Are you suggesting to combine data from both the SB ads and SB purchased product report?

Thanks again for your help

**Reply by elizafarley** [reply, marked as answer]
You can pull that same data using the [SB purchased product report](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v3/report-types/purchased-product). 

| Console report metric | API report metric |
|------|-------|
| Purchased ASIN |  [purchasedAsin](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v3/columns#purchasedAsin) | 
| Attribution type | [attributionType](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v3/columns#attributionType)| 
| 14 Day Total Sales | [sales14d](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v3/columns#sales14d)| 
| 14 Day Total Orders (#) | [orders14d](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v3/columns#orders14d) |

**Reply by rodneyo** [reply]
Eliza,
sorry but I gave you the wrong information here.  What I am actually looking for on the Sponsored purchased product report is the "promoted sku" and the "promoted asin"   These two data points are available in the report that can be run from the reporting console.

**Reply by elizafarley** [reply]
From the CSV attached in your earlier message, I don't see promoted SKU or promoted ASIN as available columns. These are not available in the API either. However, you could infer the advertised ASIN based on the `attributionType`. If the attributionType is `PROMOTED`, that should mean `purchasedAsin` is the same as the advertised ASIN.

**Comment by rodneyo** [comment]
Thanks so much Eliza,  I will give this a try

**Reply by rodneyo** [reply]
Hi again Eliza,
I'm hoping you can help me out once more on a report that our SEO staff is running from the reporting console but I can't seem to reproduce from the api.  Attached is a screen shot csv file and associated report configuration for the Sponsored Display - Purchased product.  In the excel screen shot, notice the "Purchased ASIN" column.  When I try to run the same report from the api the "purchased ASIN" is not available 

The report config sent to the api is ad_product = [SPONSORED_DISPLAY](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v3/report-types/purchased-product), report_type_id = sdPurchasedProduct.  



![CleanShot 2024-04-11 at 07 00 13@2x](https://github.com/amzn/ads-advanced-tools-docs/assets/1626333/ff9c1584-2810-4640-a1e6-244a7432f5c2)

**Reply by elizafarley** [reply]
I think the column you are looking for here is [asinBrandHalo](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v3/columns#asinbrandhalo). That will show purchased ASIN if it is different from advertised ASIN.

**Comment by rodneyo** [comment]
Also, is this type of report from the reporting console available to be reproduced from the api?
![CleanShot 2024-04-11 at 07 38 50@2x](https://github.com/amzn/ads-advanced-tools-docs/assets/1626333/48c7201e-e604-4964-b937-fa44be69cba5)

**Reply by elizafarley** [reply]
This should be equivalent to the [sbPurchasedProduct](https://advertising.amazon.com/API/docs/en-us/guides/reporting/v3/report-types/purchased-product) report.
